node/lib/util/sparse_checkout_util.js

```diff
@@ -67,23 +67,16 @@ exports.getSparseCheckoutPath = function (repo) {
 exports.inSparseMode = co.wrap(function *(repo) {
     assert.instanceOf(repo, NodeGit.Repository);
 
-    if (!(yield ConfigUtil.configIsTrue(repo, "core.sparsecheckout"))) {
-        return false;
-    }
-    let content;
-    try {
-        content = yield fs.readFile(exports.getSparseCheckoutPath(repo),
-                                    "utf8");
-    } catch (e) {
-        return false;                                                 // RETURN
-    }
-    return content === ".gitmodules\n";
+    return (yield ConfigUtil.configIsTrue(repo, "core.sparsecheckout")) ||
```

Comment:
Does configIsTrue not already return a boolean?

Reply:
If the value is not configured it returns null.

Reply:
That function, then, is probably misnamed, esp since it will usually be used in a boolean context.  But no need to fix it in this PR I guess.

Reply:
I did update the comment, which was still referencing the previous behavior.